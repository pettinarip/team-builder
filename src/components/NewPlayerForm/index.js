import NewPlayerForm from './NewPlayerForm'

export default NewPlayerForm
