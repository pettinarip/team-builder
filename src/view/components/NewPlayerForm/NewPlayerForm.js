import React from 'react'
import { Field } from 'redux-form'
import PropTypes from 'prop-types'
import SettingField from '../SettingField'

const required = value => (value ? undefined : 'Required')
const between = (min, max) =>
  (value, previousValue) => value <= max && value >= min ? value : previousValue
const maxLength = max =>
  (value, previousValue) => value.length <= max ? value : previousValue

const NewPlayerForm = ({ handleSubmit, pristine, submitting }) => (
  <form className='form-horizontal' onSubmit={handleSubmit}>
    <SettingField title='Name'>
      <Field
        autoFocus
        name='name'
        component='input'
        type='text'
        placeholder='Francesco Totti'
        validate={required}
        normalize={maxLength(30)}
      />
    </SettingField>
    <SettingField title='Number'>
      <Field
        name='number'
        component='input'
        type='number'
        placeholder='10'
        validate={required}
        normalize={between(1, 99)}
      />
    </SettingField>
    <SettingField>
      <button className='primary-button save-button button' type='submit' disabled={pristine || submitting}>
        Add to the team
      </button>
    </SettingField>
  </form>
)

NewPlayerForm.propTypes = {
  handleSubmit: PropTypes.func,
  pristine: PropTypes.bool,
  submitting: PropTypes.bool
}

export default NewPlayerForm
