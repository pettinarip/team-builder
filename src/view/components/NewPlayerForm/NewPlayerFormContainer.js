import { reduxForm } from 'redux-form'
import NewPlayerForm from './NewPlayerForm'

export default reduxForm({
  form: 'newPlayer'
})(NewPlayerForm)
