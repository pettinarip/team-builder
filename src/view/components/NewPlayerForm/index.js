import NewPlayerFormContainer from './NewPlayerFormContainer'

export default NewPlayerFormContainer
