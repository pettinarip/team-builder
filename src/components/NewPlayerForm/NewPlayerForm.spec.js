import React from 'react'
import { noop } from 'lodash'
import renderer from 'react-test-renderer'
import { createStore } from 'redux'
import {Provider} from 'react-redux'
import { reducer as formReducer } from 'redux-form'

import NewPlayerForm from './NewPlayerForm'

const mockFormStore = (initialState = {}) => {
  const store = createStore((state = initialState, action) => {
    return {
      ...state,
      form: formReducer(state.form, action)
    }
  })

  return store
}

describe('NewPlayerForm component', () => {
  it('should render correctly', () => {
    const store = mockFormStore()

    const component = renderer.create(
      <Provider store={store}>
        <NewPlayerForm onSave={noop} />
      </Provider>
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  // it('should call onSave on submit', () => {
  //   const store = mockFormStore()
  //   const onSave = jest.fn()
  //
  //   const wrapper = mount(
  //     <Provider store={store}>
  //       <NewPlayerForm onSubmit={onSave} />
  //     </Provider>
  //   )
  //
  //   wrapper.find('form').simulate('submit')
  //   expect(onSave).toHaveBeenCalled()
  // })
})
