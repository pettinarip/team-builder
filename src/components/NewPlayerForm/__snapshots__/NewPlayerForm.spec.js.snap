// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPlayerForm component should render correctly 1`] = `
<form
  className="form-horizontal"
  onSubmit={[Function]}
>
  <div
    className="settings-field-wrapper label-pad"
  >
    <label
      className="settings-field-heading"
    >
      Name
    </label>
    <div
      className="settings-field"
    >
      <input
        autoFocus={true}
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Francesco Totti"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="settings-field-wrapper label-pad"
  >
    <label
      className="settings-field-heading"
    >
      Number
    </label>
    <div
      className="settings-field"
    >
      <input
        name="number"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="10"
        type="number"
        value=""
      />
    </div>
  </div>
  <div
    className="settings-field-wrapper"
  >
    <div
      className="settings-field"
    >
      <button
        className="primary-button save-button"
        disabled={true}
        type="submit"
      >
        Add to the team
      </button>
    </div>
  </div>
</form>
`;
